packages/store/addon/-private/system/model/model.js

```diff
@@ -1363,13 +1363,34 @@ if (DEBUG) {
     return isBasicDesc(instanceDesc) && lookupDescriptor(obj.constructor, keyName) === null;
   };
 
+  const INSTANCE_DEPRECATIONS = new WeakMap();
+  const DEPRECATED_LIFECYCLE_EVENT_METHODS = [
+    'becameError',
+    'becameInvalid',
+    'didCreate',
+    'didDelete',
+    'didLoad',
+    'didUpdate',
+    'ready',
+    'rolledBack',
+  ];
+
+  let lookupDeprecations = function lookupInstanceDeprecations(instance) {
+    let deprecations = INSTANCE_DEPRECATIONS.get(instance);
+
+    if (!deprecations) {
+      deprecations = new Set();
+      INSTANCE_DEPRECATIONS.set(instance, deprecations);
+    }
+
+    return deprecations;
+  };
+
   Model.reopen({
     init() {
       this._super(...arguments);
 
-      if (DEBUG) {
```

Comment:
This is already wrapped an in `if (DEBUG) {` on https://github.com/emberjs/data/pull/6395/files#diff-f80cb1aa960a031cbf31acf375a55b0aR1343